src/interfaces/IVolatilityOracle.sol

```diff
@@ -0,0 +1,33 @@
+// SPDX-License-Identifier: BUSL 1.1
+pragma solidity 0.8.13;
+
+/**
+ * @notice This is an interface for contracts providing historical volatility,
+ * implied volatility, or both.
+ * This is used internally in order to provide a uniform way of interacting with 
+ * various volatility oracles. An external volatility oracle can be used seamlessly
+ * by being wrapped in a contract implementing this interface.
+ */
+interface IVolatilityOracle {
+    /**
+     * @notice Retrieves the historical volatility of a ERC20 token.
+     * @param token The ERC20 token for which to retrieve historical volatility.
+     * @return historicalVolatility The historical volatility of the token, scaled by 1e18
+     */
+    function getHistoricalVolatility(address token) 
+        external 
+        view 
+        returns (uint256 historicalVolatility);
+
+    /**
+     * @notice Retrieves the implied volatility of a ERC20 token.
+     * @param token The ERC20 token for which to retrieve historical volatility.
+     * @return impliedVolatility The implied volatiltiy of the token, scaled by 1e18
+     */
+    function getImpliedVolatility(address token) 
+        external 
+        view 
+        returns (uint256 impliedVolatility);
+
+    function decmimals() external view returns (uint256);
```

Comment:
What is this function about?

Reply:
a better name would be "scale"; some factor of 10 by which the getTokenYield is being scaled.

Reply:
the decimals fn is misnamed, but it's meant to return the scaling of each of the returns from the oracle.